configure.ac

```diff
@@ -75,6 +77,24 @@ AC_SEARCH_LIBS([lrint], [m], [
 ])
 pkg_libs="$LIBS"
 
+AS_IF([test "x$enable_pthreads" = xyes && test "x$ac_cv_header_stdatomic_h" != xyes], [
+    AC_MSG_ERROR([pthread support was requested, but <stdatomic.h> was not found.])
+])
```

Comment:
For Win32 targets none of those checks actually get used, yet they still run and `--enable-pthreads` can lead to configure failures while `--disable-pthreads` does nothing.

With the recent change to require `stdatomic.h` also on Win32, the configure-time header check becomes relevant for all targets. Ironically this config check is now unused on all platforms though. We can and imho should use the config-time header check instead of the compile-time macro checks. *(Consider a C implementation which doesn't provide everything to be able to declare `__STDC_VERSION__ >= 201112L` yet, but already provides `stdatomic.h`)*

Reply:
Switched to using the configure-time check for stdatomic.h.

Reply:
It still isn't possible to disable threading on Win32 builds, while it is possible *(and needed for e.g. JSO)* for all other targets via `--disable-pthreads`.
While most will probably want threading support, this disparity bugs me a bit. I am not familiar with Windows debugging tools, but perhaps a thread-less build may be useful in debugging *(and config could be generalised to `--(en|dis)able-threads`)*?